xfn-quotable/fnio.py

```diff
@@ -0,0 +1,58 @@
+import typing
+
+
+class ExplicitConnectionDetail(typing.TypedDict):
+    name: str
+    value: str
+
+
+class ObservedComposite(typing.TypedDict):
+    resource: dict
+    connectionDetails: list[ExplicitConnectionDetail]
+
+
+class ObservedResource(typing.TypedDict):
+    name: str
+    resource: dict
+    connectionDetails: list[ExplicitConnectionDetail]
+
+
+class Observed(typing.TypedDict):
+    composite: ObservedComposite
+    resources: list[ObservedResource]
+
+
+class DesiredComposite(typing.TypedDict):
+    resource: dict
+    connectionDetails: list[ExplicitConnectionDetail]
+
+
+class DesiredResource(typing.TypedDict):
+    name: str
+    resource: dict
+
+    # TODO(negz): Define these as TypedDicts.
```

Comment:
Do TypedDicts use slots? These data containers look like fixed schemas for the most part so I might have used dataclass with slots=True (and I like the other stuff they come with) but it's really just personal preference.

Reply:
I was unfamiliar with slots! This is actually my first time writing Python 3, haha. I'll look into them, thank you.